### PR TITLE
feat(mempool): add comments and unit tests for mempool, and verify the signature in api module instead of mempool.

### PR DESCRIPTION
**Pull request by Lizen0512**
feat(mempool): add comments and unit tests for mempool, and verify the signature in api module instead of mempool.

### CODE REVIEW

**Review by Alexader — APPROVED** [review]

File: pkg/order/mempool/mempool_impl.go [hunk 1]
```diff
@@ -418,14 +410,14 @@ func (mpi *mempoolImpl) processCommitTransactions(ready *raftproto.Ready) {
 	for account := range dirtyAccounts {
 		commitNonce := mpi.txStore.nonceCache.getCommitNonce(account)
 		if list, ok := mpi.txStore.allTxs[account]; ok {
-			// remove all previous seq number txs for this account.
+			// removeBySortedNonceKey all previous seq number txs for this account.
```
Comment (sandyzhou) [inline pkg/order/mempool/mempool_impl.go:413]: Should be remove?

File: pkg/order/mempool/mempool_impl.go [hunk 2]
```diff
@@ -418,14 +410,14 @@ func (mpi *mempoolImpl) processCommitTransactions(ready *raftproto.Ready) {
...
 			removedTxs := list.forward(commitNonce)
-			// remove index smaller than commitNonce delete index.
+			// removeBySortedNonceKey index smaller than commitNonce delete index.
```
Comment (sandyzhou) [inline pkg/order/mempool/mempool_impl.go:415]: Same as above

File: internal/executor/handle.go [hunk 3]
```diff
@@ -19,6 +18,8 @@ import (
 	"github.com/meshplus/bitxhub/pkg/vm"
 	"github.com/meshplus/bitxhub/pkg/vm/boltvm"
 	"github.com/meshplus/bitxhub/pkg/vm/wasm"
+
+	"github.com/cbergoon/merkletree"
```
Comment (sandyzhou) [inline internal/executor/handle.go:22]: go fmt?

File: pkg/order/mempool/utils.go [hunk 4]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/meshplus/bitxhub-model/pb"
 	"github.com/meshplus/bitxhub/internal/constant"
 	raftproto "github.com/meshplus/bitxhub/pkg/order/etcdraft/proto"
+
```
Comment (sandyzhou) [inline pkg/order/mempool/utils.go:15]: go fmt?

File: internal/executor/handle.go (hunk 3 above)
Comment (Lizen0512) [reply internal/executor/handle.go:22]: left a blank between the external and internal packages

File: pkg/order/mempool/mempool_impl.go (hunk 1 above)
Comment (Lizen0512) [reply pkg/order/mempool/mempool_impl.go:413]: yep, it's caused by renaming.

File: pkg/order/mempool/mempool_impl.go (hunk 2 above)
Comment (Lizen0512) [reply pkg/order/mempool/mempool_impl.go:415]: yep, it's caused by renaming.

File: pkg/order/mempool/utils.go (hunk 4 above)
Comment (Lizen0512) [reply pkg/order/mempool/utils.go:15]: left a blank between the external and internal package.

**Review by sandyzhou — APPROVED** [review]

**Pull request merged**